File: Base /consolidated/web/src/types/todo.ts
/**
 * TypeScript interfaces for the Todo domain.
 * All API request/response shapes are defined here.
 */

export interface Todo {
  id: number;
  title: string;
  description: string | null;
  isCompleted: boolean;
  categoryId: number | null;
  createdAt: string;
  updatedAt: string;
}

export interface TodoCreate {
  title: string;
  description?: string;
  categoryId?: number;
}

export interface TodoUpdate {
  title?: string;
  description?: string | null;
  isCompleted?: boolean;
  categoryId?: number | null;
}

File: Base /consolidated/api/app/schemas/todo.py
"""Pydantic v2 schemas for the Todo resource."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class TodoBase(BaseModel):
    """Shared fields for Todo create and update schemas."""

    title: str = Field(..., min_length=1, max_length=255, description="Todo title")
    description: str | None = Field(None, max_length=1000, description="Optional detail")
    category_id: int | None = Field(None, description="Optional category assignment")


class TodoCreate(TodoBase):
    """Schema for creating a new Todo."""
    pass


class TodoUpdate(BaseModel):
    """Schema for partially updating an existing Todo.

    All fields are optional — only provided fields will be updated.
    """

    title: str | None = Field(None, min_length=1, max_length=255)
    description: str | None = Field(None, max_length=1000)
    is_completed: bool | None = None
    category_id: int | None = None


class TodoResponse(TodoBase):
    """Schema for Todo API responses."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    is_completed: bool
    created_at: datetime
    updated_at: datetime
